### ISSUE TITLE
[Bug]: No error shown when certain bad urls is typed

### ISSUE THREAD

**Issue by hetunandu** [appsmithorg/appsmith#34846]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Adding any random route post `{pageId}/edit/` that does not match any patterns in the IDE will not lead to an error but rather an empty screen. We should handle this in some way
![Screenshot 2024-07-10 at 2 18 37 PM](https://github.com/appsmithorg/appsmith/assets/12022471/3f9b78dd-ffe1-4e01-8b56-76ca9bed8118)

### Steps To Reproduce

1. Navigate to a url that does not have a known pattern to the IDE. Check the screen and it is completely blanked

eg: https://app.appsmith.com/app/.../page1-.../edit/absolutely/random/route

### Environment

Production

### Severity

Low (Cosmetic UI issues)

### Version

Cloud

**Comment by Nikhil-Nandagopal** [comment]
@hetunandu we already have an invalid URL page right? Why can't we just use that?

**Comment by hetunandu** [comment]
@Nikhil-Nandagopal Yes we can use that. But for completely random urls i was wondering if we should redirect them to the safe url instead

**Comment by Nikhil-Nandagopal** [comment]
@hetunandu I think that becomes a slippery slope. Any invalid URL should be handled the same.
